sw/device/tests/flash_ctrl_info_access_lc.c

```diff
@@ -0,0 +1,241 @@
+
+    LOG_INFO("partition:%1d erase done", partition_number);
+    expected_irqs[kDifFlashCtrlIrqOpDone] = true;
+    expected_irqs[kDifFlashCtrlIrqProgEmpty] = true;
+    expected_irqs[kDifFlashCtrlIrqProgLvl] = true;
+    CHECK_STATUS_OK(flash_ctrl_testutils_write(
+        &flash_state, address, kPartitionId, test_data,
+        kDifFlashCtrlPartitionTypeInfo, kInfoSize));
+    compare_and_clear_irq_variables();
+  } else {
+    CHECK_STATUS_NOT_OK(flash_ctrl_testutils_write(
+        &flash_state, address, kPartitionId, test_data,
+        kDifFlashCtrlPartitionTypeInfo, kInfoSize));
+  }
+  LOG_INFO("partition:%1d write done", partition_number);
+
+  // Read task:
+  // Read page and compared with test_data.
+  uint32_t readback_data[kInfoSize];
+  if (read_allowed) {
+    expected_irqs[kDifFlashCtrlIrqOpDone] = true;
+    expected_irqs[kDifFlashCtrlIrqRdLvl] = true;
+    expected_irqs[kDifFlashCtrlIrqRdFull] = true;
+    CHECK_STATUS_OK(flash_ctrl_testutils_read(
+        &flash_state, address, kPartitionId, readback_data,
+        kDifFlashCtrlPartitionTypeInfo, kInfoSize, 1));
+    compare_and_clear_irq_variables();
+    CHECK_ARRAYS_EQ(readback_data, test_data, kInfoSize);
+
+  } else {
+    CHECK_STATUS_NOT_OK(flash_ctrl_testutils_read(
+        &flash_state, address, kPartitionId, readback_data,
+        kDifFlashCtrlPartitionTypeInfo, kInfoSize, 1));
+  }
+  LOG_INFO("partition:%1d read done", partition_number);
+}
+
+bool test_main(void) {
+  CHECK_DIF_OK(dif_lc_ctrl_init(
+      mmio_region_from_addr(TOP_EARLGREY_LC_CTRL_BASE_ADDR), &lc_ctrl));
+  CHECK_DIF_OK(dif_rv_plic_init(
+      mmio_region_from_addr(TOP_EARLGREY_RV_PLIC_BASE_ADDR), &plic0));
+
+  flash_ctrl_init_with_irqs(
+      mmio_region_from_addr(TOP_EARLGREY_FLASH_CTRL_CORE_BASE_ADDR),
+      &flash_state, &flash_ctrl);
+  rv_plic_testutils_irq_range_enable(&plic0, plic_ctx.hart_id,
+                                     kTopEarlgreyPlicIrqIdFlashCtrlProgEmpty,
+                                     kTopEarlgreyPlicIrqIdFlashCtrlOpDone);
+
+  // Enable the external IRQ at Ibex.
+  irq_global_ctrl(true);
+  irq_external_ctrl(true);
+
+  dif_lc_ctrl_id_state_t lc_id_state;
+  dif_lc_ctrl_state_t lc_state;
+  bool personalized = false;
+  // Check if device is personalized.
+  CHECK_DIF_OK(dif_lc_ctrl_get_id_state(&lc_ctrl, &lc_id_state));
+  personalized = (lc_id_state == LC_CTRL_LC_ID_STATE_STATE_VALUE_PERSONALIZED);
+  LOG_INFO("test: personalized : %d", personalized);
+
+  // Read lc state and execute info part access test.
+  // Life cycle controlled info partition access is summarized in
+  // (https://opentitan.org/book/hw/ip/lc_ctrl/doc/theory_of_operation.html#
+  // life-cycle-access-control-signals)
+  CHECK_DIF_OK(dif_lc_ctrl_get_state(&lc_ctrl, &lc_state));
+  CHECK_STATUS_OK(lc_ctrl_testutils_lc_state_log(&lc_state));
+
+  switch (lc_state) {
+    case kDifLcCtrlStateTestUnlocked0:
+      test_info_part(kFlashInfoPageIdCreatorSecret, kRandomData1, 0, 0);
```

Comment:
To improve readability 
```suggestion
      test_info_part(kFlashInfoPageIdCreatorSecret, kRandomData1, /*write_allowed=*/0, /*read_allowed=*/0);
```